### PR TITLE
readme: improve setup instructions

### PR DESCRIPTION
**Pull request by jstrnbrg**
This commit restructures the README to make it easier to understand and adds additional information for M1 Mac users.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,11 +9,12 @@ produced from the Rust code using [Rust WASM](https://rustwasm.github.io/docs/bo
 The TypeScript library is experimental. Over time, it might replace the [bitbox02-api NPM
 package](https://www.npmjs.com/package/bitbox02-api).
 
-# Rust
+## Rust (required)
```
Comment (benma) [inline README.md:12]: i find these required/optional tags confusing. typescript is required if you want to do web stuff, for example.

why not just add a note at the top of the typescript section that says "In addition to the steps described above, ..."?

File: README.md [hunk 2]
```diff
@@ -9,11 +9,12 @@ produced from the Rust code using [Rust WASM](https://rustwasm.github.io/docs/bo
...
 
-Install the protoc protobuf compiler, e.g. on Ubuntu:
+Install the protoc protobuf compiler:
 
-    sudo apt-get install protobuf-compiler
+    Ubuntu: sudo apt-get install protobuf-compiler
+    MacOS: brew install protobuf
```
Comment (benma) [inline README.md:17]: normally code/shell commands are on single lines indented by spaces so they get highlighted and are easy to copy. pls split into multiple lines, like

On Ubuntu:

    ....


On macOS:

    ....

**Review by benma — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (jstrnbrg) [reply README.md]: fixed

**Comment by jstrnbrg** [comment]
@benma PTAL

File: README.md [hunk 3]
```diff
@@ -43,6 +50,17 @@ The Rust library can be compiled to WASM package including TypeScript definition
 
 The output of this compilation will be in `./pkg`, which is a NPM package ready to be used.
 
+### M1 Macs 
+The default system clang installation currently cannot build wasm32 targets on M1 Macs. 
+Therefore a new clang compiler and archiver needs to be installed via:
+
+    brew install llvm
```
Comment (benma) [inline README.md:57]: i wonder if this is needed on non-m1 macs, i would guess so. anyway, LGTM

**Review by benma — APPROVED** [review]

**Pull request merged**